sources/packages/api/src/services/student/student.service.ts

```diff
@@ -143,27 +148,29 @@ export class StudentService extends RecordDataModelService<Student> {
         "Not able to find a student using the username (bcsc name)",
       );
     }
-
     let mustSave = false;
-
+    const userTokenBirthdate = getDateOnly(userToken.birthdate);
+    if (userToken.givenNames === undefined) {
+      userToken.givenNames = null;
+    }
     if (
-      userToken.email !== studentToSync.user.email ||
+      userTokenBirthdate.getTime() !== studentToSync.birthDate.getTime() ||
```

Comment:
Is there a reason to add the _.getTime()_ to do the comparison here?

Reply:
<img width="817" alt="image" src="https://user-images.githubusercontent.com/62901416/159047938-a44e0c52-754d-4e24-8557-35e16db705d4.png">
Yes the new Date does not work with !==  properly without the .getTime()

Reply:
![image](https://user-images.githubusercontent.com/77353155/159054113-3134acd9-916c-49d7-a147-1d604b41214b.png)
i didn't get the point

Reply:
updated to dayjs as discussed